lib/Serialization/SerializeDoc.cpp

```diff
@@ -342,7 +342,7 @@ static bool shouldIncludeDecl(Decl *D, bool ExcludeDoubleUnderscore) {
   }
 
   // Skip SPI decls.
-  if (D->isSPI())
+  if (D->getASTContext().LangOpts.EmitCommentsForSPISymbols && D->isSPI())
```

Comment:
The polarity here seems a bit odd... It looks like `false <=> skip` based on the surrounding code. So if a declaration is SPI and `EmitCommentsForSPISymbols` is true, then I would expect it to not be skipped? Is that right/am I misunderstanding something?

Reply:
You are absolutely correct! 😅 I got this backwards and wrote my tests around that behavior. Looks like i have more debugging to do...